### DISCUSSION TITLE
Unable to couple moordyn on Linux OS

### DISCUSSION THREAD

**Discussion by Guan-CH** [WEC-Sim/WEC-Sim#1255] [Q&A] [answered]
**Describe the bug**
_I attempted to download the 'libmoordyn.so' file in WEC-Sim/MoorDyn  that you added to in the past and run the program on Linux OS, but encountered an error which caused Matlab to crash.

**WEC-Sim file(s)**
_I didn't make any changes to the default file, just added the 'libmoordyn.so' file  .But I can be sure that once I add this 'libmoordyn.so' file, an error will be reported.

**To Reproduce**
1. _Add the 'libmoordyn.so' to 'WEC-Applications/Mooring/ MoorDyn'.
2. _Run the progam by type 'wecSim' in the command.
3. _See the error.

**Expected behavior**
I noticed that a few months ago, you uploaded a libmoordyn.so file in WEC-Sim/MoorDyn, but it seems that it has been deleted since then. I attempted to download and run the file on Linux OS, but encountered an error which caused Matlab to crash. 
I would appreciate it if you could provide me with more details about the specific reason for the error in your libmoordyn.so file, as well as any instructions on how to successfully run moordyn on Linux.

**Screenshots**
![matlab_error](https://github.com/WEC-Sim/WEC-Sim/assets/166460766/66134237-a739-4253-867c-6159f84077f2)

**Workstation Information (this section MUST BE completed):**
 - OS: Linux 5.15.0
 - [MATLAB/Simulink Release](https://www.mathworks.com/help/matlab/release-notes.html) [ 2020b]
 - [WEC-Sim Release](http://wec-sim.github.io/WEC-Sim/man/release_notes.html) [ v6.0]

**Additional context**
_N/A_

**Comment by jtgrasb** [comment]
Thank you for your question. We ask users to open GitHub issues using an appropriate template. This is helpful for the WEC-Sim team to classify issues, and address questions more thoroughly and efficiently. We hope this will be beneficial for developers, users and the WEC-Sim community in general.

Can you please edit this issue to use the appropriate issue template and include more detail about your setup and error message? Thank you.

**Comment by Guan-CH** [comment]
Okay, I've reedited my issue. Thank you for taking time out of your busy schedule to answer my question！ @jtgrasb

**Comment by jtgrasb** [comment, marked as answer]
Thank you for your question and using the template, @Guan-CH. The .so file you downloaded which was added to the WEC-Sim/MoorDyn repository previously was removed because it was causing the tests to fail, Unfortunately, the WEC-Sim team does not have direct access to a linux machine, so we were relying on the MoorDyn tool to generate the .so file. It seems the generated .so file did not work, so we chose to pause support for WEC-Sim+MoorDyn on Linux until we can reliably generate a new .so file that works. We are working on addressing this issue currently but have not resolved it yet.

At present, I recommend using the [compiling instructions](https://moordyn.readthedocs.io/en/latest/compiling.html#) on the MoorDyn Documentation. This gives multiple options including pre-compiled releases, compiling as a simple library, and compiling using cmake. We have been working on using the pre-compiled release to generate the .so file, but have had issues with this so far, so you may need to use one of the other options. I apologize for the inconvenience and we hope to be able to support WEC-Sim+MoorDyn in the future without requiring manual compiling. If you are able to compile it, it would be much appreciated if you shared the .so file so we could add this capability back to WEC-Sim for other users. Otherwise, I expect that we will be able to support this feature in the next few months after resolving the compiling issues.

**Comment by Guan-CH** [comment]
Thanks for your careful answers and suggestions, I will try using MoorDyn to see if I can generate the required .so file  in another way.

**Comment by jtgrasb** [comment]
HI @Guan-CH, we are still working on the WEC-Sim+MoorDyn updates at the moment and have not explored the Linux compatibility yet. I am converting this issue to a discussion to keep it open. We look forward to hearing any updates on generating the .so file.

**Reply by Guan-CH** [reply]
Thank you very much for your team's efforts on this! However, I'm sorry that I still haven't been able to resolve this issue myself.
